test/expect/TestScript.test_onnx_export_script_module_if.expect

```diff
@@ -6,20 +6,21 @@ ModelProto {
     GraphProto {
       name: "torch-jit-export"
       inputs: [{name: "x.1", type:Tensor dims: 1 2 3}]
-      outputs: [{name: "4", type:Tensor dims: 1 2 3}]
+      outputs: [{name: "5", type:Tensor dims: 1 2 3}]
       initializers: []
       nodes: [
         Node {type: "ReduceSum", inputs: [x.1], outputs: [1], attributes: [{ name: 'keepdims', type: int, value: 0}]},
         Node {type: "Constant", inputs: [], outputs: [2], attributes: [{ name: 'value', type: tensor, value:TensorProto shape: []}]},
-        Node {type: "Greater", inputs: [1,2], outputs: [3], attributes: []},
-        Node {type: "If", inputs: [3], outputs: [4], attributes: [{ name: 'then_branch', type: graph, value:
+        Node {type: "ATen", inputs: [2,1], outputs: [3], attributes: [{ name: 'operator', type: string, value: 'type_as'}]},
```

Comment:
Yes, I believe so. The form `expr > 0` seems pretty common and would be expected to work for anyone used to implicit conversions. Can we solve this restricted case by doing a peephole optimization that matches (Constant -> type_as) and cast the Constant node to the expected type? If `expr` is dynamic, we're probably screwed though

Reply:
Thanks for the confirmation, @jamesr66a.  For now I think I could eliminate the aten::type_as nodes that I added right before exporting to onnx so that the behavior before this PR is unchanged.